GDSCommon-Demo/GDSCommon-Demo/Mocks/MockListViewModel.swift

```diff
@@ -0,0 +1,29 @@
+import GDSCommon
+
+struct MockListViewModel: ListOptionsViewModel {
+    let title: GDSLocalisedString = "This is the List Options screen pattern"
+    let body: String? = "This is the optional body label. If the view model property is `nil` then the label is hidden."
+    let listRows: [GDSLocalisedString] = ["Table view list item 1", "Table view list item two", "Table view list item 3", "Table view list item IV"]
+    let listFooter: String? = "Optional footer. Configure it on the view model in a similar way as the `body` property. The right bar button works the same way."
+    let buttonViewModel: ButtonViewModel = MockButtonViewModel(title: "Action button",
+                                                                      action: { print("button was tapped") })
+    let resultAction: (GDSLocalisedString) -> Void
+    let rightBarButtonTitle: GDSLocalisedString? = "Right bar button"
+
+    let screenView: () -> Void
+    
+    func didDismiss() {
+        print("did dismiss")
```

Comment:
More a question than a comment/request - do we want to have print statements in places like this? I know that its there, in a mock, just to show it works but wondering if this is the best way?

Reply:
Yeah, good thing to discuss. The `didDismiss` method is hard to test with unit tests because it is typically triggered by the `rightBarButtonItem` which I've yet to find a way to programmatically trigger the action for. Do you know a way?

If we were to add some UITests we could tap the button and then test the effect. So I left the print statement in there so users of the mock can see _something_ even if it cannot be programmatically tested.

Reply:
We've triggered a bar button item programmatically in tests, checkout the `Tests/Screens/Permissions/AnalyticsPermissionViewControllerTests` file and specifically the `test_rightBarButtonItem_Contents` test which has:
```swift
_ = rightBarButton?.target?.perform(rightBarButton?.action, with: nil)
```
Not sure if this is best practice or not but seems to work, just to mention 🙂

Reply:
Hmm, I'd tried that. Will have another look though.

Reply:
It works now so I guess it did just need the host app :) Thanks for the comments @DylanRJ and @benmckillop